Client.Core/Flux/Internal/FluxResultMapper.cs

```diff
@@ -106,85 +110,67 @@ private void SetFieldValue<T>(T poco, PropertyInfo property, object value)
                     return;
                 }
 
-                if (typeof(int).IsAssignableFrom(propertyType))
+                if (propertyType == typeof(Instant))
                 {
-                    property.SetValue(poco, ToIntValue(value));
+                    property.SetValue(poco, ToInstantValue(value));
                     return;
                 }
 
-                if (typeof(bool).IsAssignableFrom(propertyType))
+                if (value is IConvertible)
                 {
-                    property.SetValue(poco, bool.TryParse(value.ToString(), out var v) && v);
-                    return;
+                    // Nullable types cannot be used in type conversion, but we can use Nullable.GetUnderlyingType()
+                    // to determine whether the type is nullable and convert to the underlying type instead
+                    Type targetType = Nullable.GetUnderlyingType(propertyType) ?? propertyType;
+                    property.SetValue(poco, Convert.ChangeType(value, targetType));
                 }
-
-                if (typeof(DateTime).IsAssignableFrom(propertyType))
+                else
                 {
-                    property.SetValue(poco, ToDateTimeValue(value));
-                    return;
+                    property.SetValue(poco, value);
                 }
-
-                property.SetValue(poco, value);
             }
-            catch (InvalidCastException)
+            catch (InvalidCastException ex)
             {
                 throw new InfluxException(
-                    $"Class '{poco.GetType().Name}' field '{property.Name}' was defined with a different field type and caused a InvalidCastException. " +
-                    $"The correct type is '{value.GetType().Name}' (current field value: '{value}').");
+                    $"Class '{poco.GetType().Name}' field '{property.Name}' was defined with a different field type and caused an exception. " +
+                    $"The correct type is '{value.GetType().Name}' (current field value: '{value}'). Exception: {ex.Message}", ex);
             }
         }
 
-        private double ToDoubleValue(object value)
+        private DateTime ToDateTimeValue(object value)
         {
-            if (value.GetType().IsAssignableFrom(typeof(double)))
+            if (value is DateTime dateTime)
             {
-                return (double) value;
+                return dateTime;
             }
 
-            return (double) value;
-        }
-
-        private long ToLongValue(object value)
-        {
-            if (value.GetType().IsAssignableFrom(typeof(long)))
+            if (value is Instant instant)
             {
-                return (long) value;
+                return instant.InUtc().ToDateTimeUtc();
             }
 
-            return (long) value;
+            return (DateTime) value;
         }
 
-        private int ToIntValue(object value)
+        private Instant ToInstantValue(object value)
```

Comment:
The following statement is unnecessary in `ToInstantValue`:

```csharp
if (value is Instant instant)
{
     return instant;
}
```

Reply:
I do not feel this is true.  poco property may be `Instant`, but object value coming may not be `Instant`

Reply:
(see comment above for justification)